src/client/app/components/IntervalControlsComponent.tsx

```diff
@@ -0,0 +1,169 @@
+		// Do not allow bad values in state.
+		const isCustom = !(['1', '7', '28'].find(days => days == duration.asDays().toString()));
+		setShowCustomDuration(isCustom);
+		setDaysCustom(duration.asDays());
+		setDays(isCustom ? CUSTOM_INPUT : duration.asDays().toString());
+	}, [duration]);
+
+	// Returns true if this is a valid duration.
+	const daysValid = (days: number) => {
+		return Number.isInteger(days) && days >= MIN_DAYS && days <= MAX_DAYS;
+	};
+
+	// Updates values when the standard duration menu for bar or map is used.
+	const handleDaysChange = (value: string) => {
+		if (value === CUSTOM_INPUT) {
+			// Set menu value from standard value to special value to show custom
+			// and show the custom input area.
+			setDays(CUSTOM_INPUT);
+			setShowCustomDuration(true);
+		} else {
+			// Set the standard menu value, hide the custom duration input
+			// and duration for graphing.
+			// Since controlled values know it is a valid integer.
+			setShowCustomDuration(false);
+			updateDurationChange(Number(value));
+		}
+	};
+
+	// Updates value when the custom duration input is used for bar or map.
+	const handleCustomDaysChange = (value: number) => {
+		setDaysCustom(value);
+	};
+
+	const handleEnter = (key: string) => {
+		// This detects the enter key and then uses the previously entered custom
+		// duration to set the duration for the graphic.
+		if (key === 'Enter') {
+			updateDurationChange(daysCustom);
+		}
+	};
+
+	const updateDurationChange = (value: number) => {
+		// Update if okay value. May not be okay if this came from user entry in custom form.
+		if (daysValid(value)) {
+			dispatch(graphSlice.actions.updateDuration(moment.duration(value, 'days')));
+		}
+	};
+
+	// Updates values when the compare period menu is used.
+	const handleComparePeriodChange = (value: string) => {
+		if (chartType === ChartTypes.compare) {
+			const period = value as unknown as ComparePeriod;
+			dispatch(graphSlice.actions.updateComparePeriod({ comparePeriod: period, currentTime: moment() }));
+		}
+	};
+
+	return (
+		<div>
+			<div style={divTopBottomPadding}>
+				<p style={labelStyle}>
+					{(chartType === ChartTypes.bar && translate('bar.interval')) ||
+						(chartType === ChartTypes.map && translate('map.interval')) ||
+						(chartType === ChartTypes.compare && translate('compare.period'))}:
+					<TooltipMarkerComponent page='home' helpTextId={
+						chartType === ChartTypes.bar ? 'help.home.bar.days.tip' :
+							chartType === ChartTypes.map ? 'help.home.map.days.tip' :
+								'help.home.compare.period.tip'
+					} />
+				</p>
+				<Input
+					id='durationDays'
+					name='durationDays'
+					type='select'
+					value={chartType === ChartTypes.compare ? comparePeriod?.toString() : days}
+					onChange={e => chartType === ChartTypes.compare ? handleComparePeriodChange(e.target.value) : handleDaysChange(e.target.value)}
+				>
+					{chartType !== ChartTypes.compare && (
```

Comment:
I think this would be cleaner if the same three options were added and then custom is added if it is not compare. Let me know if you want help with the way to do this.